File: C#/0762. Prime Number of Set Bits in Binary Representation.cs
public class Solution {
    public int CountPrimeSetBits(int L, int R) {
        int[] nums=new int[]{2,3,5,7,11,13,17,19,23,29};
        HashSet<int> primes=new HashSet<int>(nums);
        int cnt=0;
        for(int i=L;i<=R;i++){
            if(IsPrimes(i,primes)){
                cnt++;
            }
        }
        return cnt;
    }
    public bool IsPrimes(int i,HashSet<int> primes){
        int rep=0;
        while(i!=0){
            rep+=i&1;
            i=i>>1;
        }
        return primes.Contains(rep);
    }
}

File: 20190519python/0762. Prime Number of Set Bits in Binary Representation.py
class Solution:
    def countPrimeSetBits(self, L: int, R: int) -> int:
        rep=0
        for num in range(L,R+1):
            cnt=self.count1(num)
            if self.isPrime(cnt):
                rep+=1
        return rep

    def count1(self,num):
        rep=0
        while num>0:
            rep+=num&1
            num>>=1
        return rep

    def isPrime(self,num):
        rep=(2,3,5,7,11,13,17,19,23,29,31)
        if num in rep:
            return True
        return False
